.github/workflows/tests.yaml

```diff
@@ -336,6 +416,85 @@ jobs:
           dag_tarball_version_after: ${{ steps.get-deployment-after.outputs.desiredDagTarballVersion }}
           image_version_after: ${{ steps.get-deployment-after.outputs.desired_image_version }}
 
+  # DBT Deploy test would test the DAG only deploy functionality in deploy action
+  dbt-deploy-test:
+    name: DBT Deploy Test
+    runs-on: ubuntu-latest
+    needs: [custom-docker-image-test, create-test-deployments]
```

Comment:
it does not depend on the `custom-docker-image-test`, but because these tests are running against a single deployment, to avoid any flakiness in test results, we are running a single test at a time, and hence we are required to use the `needs` clause to create the dependencies to avoid concurrent test scenario runs